subprojects/core/src/integTest/groovy/org/gradle/api/internal/tasks/TaskCacheabilityReasonIntegrationTest.groovy

```diff
@@ -43,20 +43,26 @@ class TaskCacheabilityReasonIntegrationTest extends AbstractIntegrationSpec impl
         buildFile """
             import org.gradle.api.internal.tasks.TaskOutputCachingDisabledReasonCategory
 
-            class NotCacheable extends DefaultTask {
+            class UnspecifiedCacheabilityTask extends DefaultTask {
                 @Input
                 String message = "Hello World"
                 @OutputFile
                 File outputFile = new File(temporaryDir, "output.txt")
 
                 @TaskAction
-                public void generate() {
+                void generate() {
                     outputFile.text = message
                 }
             }
 
+            @DisableCachingByDefault
+            class NotCacheableByDefault extends UnspecifiedCacheabilityTask {}
+
+            @DisableCachingByDefault(because = 'do-not-cache-by-default reason')
```

Comment:
```suggestion
            @DisableCachingByDefault(because = 'disable-caching-by-default reason')
```

Reply:
I don't think it's a goal to unify these. You could also argue that `NotCacheableByDefaultWithReason` should be `CachingDisabledByDefaultWithReason` etc. I think it's fine to have these be descriptive names but not 100% the same.

Reply:
I would prefer if you at least change the reason her, because `do-not-cache-by-default` is gone now. I'll survive if you don't.